### ISSUE TITLE
Perform weight update during validation

### ISSUE THREAD

**Issue by yiqingguobuaa** [SKrisanski/FSCT#12]
Hi Sean,

Thanks for your reply to my previous thread and sorry to bother you again.

I am just curious about the following lines:

https://github.com/SKrisanski/FSCT/blob/7f1bd48a6dd9e86dc4333aa6029749514e303637/scripts/train.py#L337-L339

I am not experienced with Pytorch coding, but from what I've seen in other people's codes, they normally do not perform *optimizer.step()* during the validation stage. I think the idea is to learn from the training set only, while keeping the validation set as independent as possible, so that it can provide an objective judgement on model performance in each epoch. 

Is there a reason that we would perform *optimizer.step()* here? 

Many thanks and wish you a nice weekend.

Cheers, Yiqing.

**Comment by SKrisanski** [comment]
Hi Yiqing,

You are correct, it should not be doing this. In practice, it shouldn't matter too much, as this is just guiding information during the training process, but this is certainly a bug. I've just pushed a fix.

Thanks for letting me know!
Sean
